### ISSUE TITLE
Feature Request: Indicate "secure" environment variable

### ISSUE THREAD

**Issue by ravensorb** [nuxxapp/nuxx#26]
As a thought for the feature, it might be very useful to be able to indicate if an env variable is coming from a secret file and/or should a .env be generated rather than putting everything in the compose file.

**Comment by corpulent** [comment]
@ravensorb I have been thinking about this.  A few users requested a feature similar to what you are asking.  It will work this way:  a user logs has to log in first, and then they can define a global variable set for a project.  This can act as both, a `.env` file with its contents to be exported,  and as variable placeholders to be used in the forms.  For example, using `{{DB_PASS}}` placeholder within the forms, so when the docker-compose.yml is generated with that placeholder,  it grabs the value from the global variable set.  Let me know what you think of this.

**Comment by basudip** [comment]
not only env variable, but it can be implemented on volume mount also. a fixed set of volume mount like 
/media/mount/Docker/
/media/mount/TV/
/media/mount/Movie/
/media/mount/Download/

hope you get the idea.

**Comment by corpulent** [comment]
@basudip yea I totally get it.  Ill drop a wireframe here tomorrow morning to gather some more thoughts before building it out.

**Comment by ravensorb** [comment]
Love it - might be nice to be able to set them at an admin/global level too?

**Comment by corpulent** [comment]
@ravensorb yea thats the idea.  "Project level", "Global level".

**Comment by corpulent** [comment]
@ravensorb @basudip Here is my idea of global and project level variables, let me know what you think.  Reading env vars from `.env` file falls into the recipes work,  I need more time to think about this.

<img width="905" alt="Screen Shot 2020-06-24 at 2 38 40 PM" src="https://user-images.githubusercontent.com/849403/85613944-694fb580-b628-11ea-8a4f-fa7d8e546c5a.png">

**Comment by ravensorb** [comment]
I think this makes a lot of sense.  

Thinking this through "out load" 
- An admin can come in and add a global variable
- An admin can leverage  a global variable as a token/ref within a template (volume, network, env, etc)
- A user can add a user variable
- A user can leverage a global or user variable as a token/ref within a template (volume, network, env, etc)
- These referenced variables would be "expanded" when the template is generated (env file or compose file)

Sound right?

Also couple of questions-- is it possible to reference a global var in a token for a project/user var?  or would it be possible to "override" a global var at a project level?  Note: This might be overkill, I am just curious on the options

**Comment by basudip** [comment]
yes, this makes absolute sense. create a selectable tag/label/icon where user/admin can define what type of variable it will be. like ENV variable or volume path etc.
